core/src/test/scala/org/apache/spark/sql/delta/test/ScanReportHelper.scala

```diff
@@ -0,0 +1,163 @@
+          case NonFatal(e) =>
+            logDebug("Not running Delta Metering because the query failed during analysis.", e)
+            return
+        }
+
+        val fileScans = collectScans(qe.executedPlan)
+
+        for (scanExec <- fileScans) {
+          scanExec.relation.location match {
+            case deltaTable: PreparedDeltaFileIndex =>
+              val preparedScan = deltaTable.preparedScan
+              // The names of the partition columns that were used as filters in this scan.
+              // Convert this to a set first to avoid double-counting partition columns that might
+              // appear multiple times.
+              val usedPartitionColumns =
+              preparedScan.partitionFilters.map(_.references.map(_.name)).flatten.toSet.toSeq
+              val report = ScanReport(
+                tableId = deltaTable.deltaLog.snapshot.metadata.id,
+                path = deltaTable.path.toString,
+                scanType = "delta-query",
+                deltaDataSkippingType = preparedScan.dataSkippingType.toString,
+                partitionFilters = preparedScan.partitionFilters.map(_.sql).toSeq,
+                dataFilters = preparedScan.dataFilters.map(_.sql).toSeq,
+                unusedFilters = preparedScan.unusedFilters.map(_.sql).toSeq,
+                size = Map(
+                  "total" -> preparedScan.total,
+                  "partition" -> preparedScan.partition,
+                  "scanned" -> preparedScan.scanned),
+                metrics = scanExec.metrics.mapValues(_.value).toMap +
+                  ("scanDurationMs" -> preparedScan.scanDurationMs),
+                annotations = Map.empty, // TODO(scott): annotations
+                versionScanned = deltaTable.versionScanned,
+                usedPartitionColumns = usedPartitionColumns,
+                numUsedPartitionColumns = usedPartitionColumns.size,
+                allPartitionColumns = deltaTable.deltaLog.snapshot.metadata.partitionColumns,
+                numAllPartitionColumns = deltaTable.deltaLog.snapshot.metadata.partitionColumns.size
+              )
+
+              scans += report
+
+            case deltaTable: TahoeFileIndex =>
+              val report = ScanReport(
+                tableId = deltaTable.deltaLog.snapshot.metadata.id,
+                path = deltaTable.path.toString,
+                scanType = "delta-unknown",
+                partitionFilters = Nil,
+                dataFilters = Nil,
+                unusedFilters = Nil,
+                size = Map(
+                  "total" -> DataSize(
+                    bytesCompressed = Some(deltaTable.deltaLog.snapshot.sizeInBytes))
+                ),
+                metrics = scanExec.metrics.mapValues(_.value).toMap,
+                versionScanned = None,
+                annotations = Map.empty // TODO(scott): annotations
+              )
+
+              scans += report
+
+            case _ if scanExec.relation.location.rootPaths.nonEmpty =>
```

Comment:
what is this case for? would delta always have TahoeFileIndex? or are you planning this to be generic to non delta tables too?

Reply:
this case covers `parquet` tables, for example.

Reply:
Is this is needed for this PR? if not, i suggest you remove it. unnecessary code add clutter... we can always add this if needed for parquet in the future.